report.go

```diff
@@ -1,106 +1,157 @@
 package gomoss
+
 import (
-    "github.com/anaskhan96/soup"
-    "strings"
-    "os"
-    "regexp"
-    "fmt"
+	"fmt"
+	"github.com/anaskhan96/soup"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
 )
 
-// Result of extracted data from Moss URLs
-type Result struct {
-    Codes []string
-    Percentages []string
-    Similarities []string
+// Region specifies similarity from line a to line b
+type Region struct {
+	from int64
+	to   int64
+}
+
+// Source is structure that containing plagiarism result of a source code
+type Source struct {
+	Filename   string
+	Similarity string
+	Regions    []Region
+}
+
+// Match means compare the plagiarism between two source codes
+type Match struct {
+	Left, Right Source
+}
+
+// Report is data extracted data from Moss URL
+type Report struct {
+	Matches []Match
 }
 
-// FindMatches find all analysis related hrefs in HTML
-func FindMatches(baseURL string) []string {
-    resp, _ := soup.Get(baseURL)
-
-    doc := soup.HTMLParse(resp)
-    anchors := doc.FindAll("a")
-    exist := make(map[string]bool)
-    var urls []string
-    for _, anchor := range anchors {
-        url := anchor.Attrs()["href"]
-        if strings.Contains(url, "match") && !exist[url] {
-            exist[url] = true
-            urls = append(urls, url)
-        }
-    }
-    return urls
+// FindLinks find all analysis related hrefs in HTML
+func findLinks(baseURL string) []string {
+	resp, _ := soup.Get(baseURL)
```

Comment:
nit: every call that returns an `error`, you should use something with it. typically bubble it up.